reference/api/batches.mdx

```diff
@@ -0,0 +1,244 @@
+---
+title: Batches — Meilisearch API reference
+description: The /batches route allows you to monitor how Meilisearch is grouping and processing asynchronous operations.
+---
+
+# Batches
+
+The `/batches` route gives information about the progress of batches of [asynchronous operations](/learn/async/asynchronous_operations).
+
+## Batch object
+
+```json
+{
+  "uid": 0,
+  "details": {
+    "receivedDocuments": 6,
+    "indexedDocuments": 6
+  },
+  "stats": {
+    "totalNbTasks": 1,
+    "status": {
+      "succeeded": 1
+    },
+    "types": {
+      "documentAdditionOrUpdate": 1
+    },
+    "indexUids": {
+      "INDEX_NAME": 1
+    }
+  },
+  "duration": "PT0.250518S",
+  "startedAt": "2024-12-10T15:20:30.18182Z",
+  "finishedAt": "2024-12-10T15:20:30.432338Z",
+  "progress": {
+    "currentStep": [
+      { 
+        "name": "extracting words",
+        "finished": 2,
+        "total": 9,
+      },
+      {
+        "name": "document",
+        "finished": 30546,
+        "total": 31944,
+      }
+    ],
+    "percentage": 32.8471
+  }
+}
+```
+
+### `uid`
+
+**Type**: Integer<br />
+**Description**: Unique sequential identifier of the batch. Starts at `0` and increases by one for every new patch.
+
+### `details`
+
+**Type**: Object<br />
+**Description**: Basic information on the types tasks in a batch. Consult the [task object reference](/reference/api/tasks#details) for an exhaustive list of possible values.
+
+### `progress`
+
+**Type**: Object<br />
+**Description**: Object containing two fields: `currentStep` and `percentage`. Once Meilisearch has fully processed a batch, its `progress` is set to `null`.
+
+#### `currentStep`
+
+Information about the current operations Meilisearch is performing in this batch. A step may consist of multiple substeps.
+
+| Name            | Description                                        |
+| :---------------| :------------------------------------------------- |
+| **`name`**      | A string describing the operation                  |
+| **`total`**     | The total number of operations in the step         |
+| **`finished`**  | The number of operations Meilisearch has completed |
+
+#### `percentage`
+
+The progress of the current set of operations, calculated from all current steps and substeps.
```

Comment:
Thanks for the suggestion. I'll push an update stating `percentage` does not grow linearly, and added a tip to `steps` advising users to monitor the last step's `finished` in case a batch is taking longer than expected.